src/extension.ts

```diff
@@ -93,6 +93,27 @@ export function activate(context: ExtensionContext) {
         runTextInTerm(callableTerminal, wrappedText);
     }
 
+    async function runCommandWithSelectionOrWord(rCommand: string) {
+        const text = getWordOrSelection().join("\n");
+        const callableTerminal = await chooseTerminal();
+        let call = rCommand.replace("$1", text);
+        runTextInTerm(callableTerminal, [call]);
+    }
+
+    async function runCommandWithEditorPath(rCommand: string) {
+        const wad: TextDocument = window.activeTextEditor.document;
+        wad.save();
+        const callableTerminal = await chooseTerminal();
+        let rPath = ToRStringLiteral(wad.fileName, '"');
```

Comment:
I suggest we use empty quote here (e.g. `ToRStringLiteral(wad.fileName, "")`) to be consistent with `runCommandWithSelectionOrWord` which does not quote the selection or word, and this also allows the use case of the following:

```json
{
        "description": "write file",
        "key": "ctrl+shift+w",
        "command": "r.runCommandWithEditorPath",
        "when": "editorTextFocus",
        "args": "writeLines('title: $1', '$1')"
}
```

Reply:
I was 50-50 on this. Consistency was a strong pull, but it seemed like since it would be the full path with separators and extension, you're very unlikely to use it other than as a string literal. But yeah happy to change.